api/v1beta2/cryostat_types.go

```diff
@@ -344,7 +344,12 @@ type StorageServiceConfig struct {
 	// HTTP port number for the storage to be created by the operator.
 	// Defaults to 8333.
 	// +optional
-	HTTPPort      *int32 `json:"httpPort,omitempty"`
+	HTTPPort *int32 `json:"httpPort,omitempty"`
+	// HTTPS port number for the storage to be created by the operator.
+	// Only used if TLS is enabled.
+	// Defaults to 8334.
+	// +optional
+	HTTPSPort     *int32 `json:"httpsPort,omitempty"`
```

Comment:
What about using `HTTPPort` whether TLS is enabled or not? For example, with Reports, we have just the one HTTPPort property.

Reply:
I thought about that, but it doesn't seem that SeaweedFS' S3 implementation allows for completely turning off the HTTP port. HTTPS can only be turned on as an additional port. So, I figured I would reflect that here. I can remove it from the API and just "hide"/ignore the HTTP port internally when HTTPS is enabled, though, if you think that'd be better.

Reply:
Hiding it might be better. We could remove it from the Service, and also block traffic to it using the network policy?

Reply:
I might be misreading this, but it looks like if the HTTPS port isn't provided, but the private key is, it will serve HTTPS over the configured HTTP port:
https://github.com/seaweedfs/seaweedfs/blob/95357df8dda57baba53b15dad6efcb6be9a2c123/weed/command/s3.go#L338-L367

Reply:
Nice find! I tried something like that with just `podman run` and the container image, but I think I only tried setting the HTTPS port to a blank value. This way seems to work well.